### PR TITLE
Changed  CI docker from gitlab registry to public dockerhub account

### PR DESCRIPTION
**Pull request by tanjpg**
For security purposes, the docker Image was changed from the ROS Industrial Asia Pacific gitlab registry to a public Dockerhub account for both the gitlab-ci.yml and travis.yml files

### CODE REVIEW

**Comment by cardboardcode** [comment]
For the package.xml file found [here](https://github.com/tanjpg/easy_manipulation_deployment/blob/f6fb76df045aad3e12fb0d234fbbaa67be800cad/assets/environment/table_description/package.xml#L7),

Please include the **maintainer email** and **License** information.

**Comment by cardboardcode** [comment]
May I enquire, since the documentation is already hosted externally [here]() as indicated by the root directory  `README.md`, 

Is there still a need to keep the [docs](https://github.com/tanjpg/easy_manipulation_deployment/tree/master/docs) folder in the repository?

**Reason**: Removing the **docs** folder will help minimize the amount of time required to clone the repository onto a local workstation and improve user experience. However, if there is strong rationale to keep it, we can just keep it then.

**Comment by cardboardcode** [comment]
1. Please remove the **.vcode** folder found [here](https://github.com/tanjpg/easy_manipulation_deployment/tree/master/grasp_planner/.vscode).

2. Please create a  `.gitignore` file in the root directory and add the following line to untrack the aforementioned folder.
```
easy_manipulation_deployment/grasp_planner/.vscode/*
```

**Reason**: This folder is likely only used for a developer's personalized ide settings using Visual Code and probably not meant to be shared. This also cuts down on unnecessary files in the repository.

**Comment by cardboardcode** [comment]
Please remove the **lcov** folder found [here](https://github.com/tanjpg/easy_manipulation_deployment/tree/master/grasp_planner/lcov). 

**Reason**: The code coverage report can be found on codecov.com and accessible by clicking on the code coverage badge in the root  `README.md`. If there is a strong rationale to keep it, do provide it. Otherwise, it will be beneficial to remove unnecessary files from the repository.

**Comment by cardboardcode** [comment]
1. Please remove the **__pycache__** folder found [here](https://github.com/tanjpg/easy_manipulation_deployment/tree/master/grasp_validator/grasp_validator/__pycache__).

2. Add the following line in the `.gitignore` file you have created earlier based on an earlier requested change in this thread.

 ```
*.pyc
```

**Reason**: The **.pyc** files are transitional which means that they need not be included in the repository and is only utilized to help run the python code faster on a specific workstation that it has been compiled on. Similar benefit to the above requested change.

**Comment by cardboardcode** [comment]
May I enquire if the following **empty file** is still needed?

The file in question is [/grasp_validator/resource/grasp_validator](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/grasp_validator/resource/grasp_validator).

**Reason**: If it is not needed, please delete it along with the **resources** folder that contains this only file. Same benefit as above.

**Comment by cardboardcode** [comment]
Please remove the __pycache__ folder found [here](https://github.com/tanjpg/easy_manipulation_deployment/tree/master/grasp_validator/test/__pycache__) as well.

 **Reason**: Same reason for the previously mentioned pycache folder.

**Comment by cardboardcode** [comment]
The following are requested edits for files under the folder, [scenes](https://github.com/tanjpg/easy_manipulation_deployment/tree/master/scenes).

1. Please include the **copyright header** for the python script [scenes/fanuc_test/launch/demo.launch.py](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/scenes/fanuc_test/launch/demo.launch.py).

2. Please do the same for the python script [scenes/panda_test/launch/demo.launch.py](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/scenes/panda_test/launch/demo.launch.py).

3. Please do the same for the python script [scenes/ur10_test/launch/demo.launch.py](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/scenes/ur10_test/launch/demo.launch.py).

4. Please do the same for the python script [scenes/ur3_test/launch/demo.launch.py](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/scenes/ur3_test/launch/demo.launch.py).

5. Please do the same for the python script [scenes/ur5_test/launch/demo.launch.py](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/scenes/ur5_test/launch/demo.launch.py).

**Reason**: To reserve the appropriate ownership and licensing details for this file under the right authorship, according to REP 2004.

**Comment by gavanderhoorn** [comment]
[github/gitignore](https://github.com/github/gitignore) has a collection of "standard" `.gitignore` files which contains often used patterns. There's one for Python development, C++, etc.

You are free to combine them of course. And `.gitignore` files can be added to sub directories instead of the root of the repository as well, which may make managing the paths in them a little easier.

**Comment by cardboardcode** [comment]
Please include the copyright header for the python script [workcell_builder/examples/ros2/launch/demo.launch.py](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/workcell_builder/examples/ros2/launch/demo.launch.py).

**Reason**: Same reason as above regarding licensing.

**Comment by cardboardcode** [comment]
Please include the following changes to the root `README.md` found [here](https://github.com/tanjpg/easy_manipulation_deployment/blob/master/README.md).

1. Change the following the phrasing as shown below from:
```
easy to deploy 
pick and place
```
to 
```
easy-to-deploy 
pick-and-place
```

**Reason**: The phrases used suggest that they are meant to be a single noun and not a combination of separate verbs/adjectives. This requested change is subjective and therefore only recommended, not enforced. 

2. Please remove the leading 2 asterisks found in this abstract of the root `README.md`.

**Note that this package is currently in its Alpha release, therefore many components are currently under heavy development and are subject to change.

**Reason**: The 2 leading asterisks seem to be a formatting error and the asterisks needed to bold the entire statement in question are already there.

**Comment by tanjpg** [comment]
@cardboardcode made the necessary changes except for issues regarding documentation/naming issues as well as code coverage folders. will fix those in a separate PR

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+easy_manipulation_deployment/grasp_planner/.vscode/*
+easy_manipulation_deployment/workcell_builder/workcell_builder/include/attributes/.vscode/*
```
Comment (Briancbn) [inline .gitignore:2]: Combine vscode ignore.
```suggestion
.vscode/
```

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+easy_manipulation_deployment/grasp_planner/.vscode/*
```
Comment (Briancbn) [inline .gitignore:1]:
```suggestion
```

**Review by Briancbn — CHANGES_REQUESTED** [review]

**Comment by cardboardcode** [comment]
> For the package.xml file found [here](https://github.com/tanjpg/easy_manipulation_deployment/blob/f6fb76df045aad3e12fb0d234fbbaa67be800cad/assets/environment/table_description/package.xml#L7),
> 
> Please include the **maintainer email** and **License** information.
> 
> Under the same directory of this file found [here](https://github.com/tanjpg/easy_manipulation_deployment/tree/master/assets/environment/table_description), please include a `LICENSE` file that contains all necessary boilerplate licensing information for the license you have chosen.
> 
> **Reason**: This is to ensure the appropriate copyright has displayed for files that is written by the rightful code author. This is also in adherence to REP 2004 quality assurance standards.

@tanjpg @Briancbn Please address the following requested change.

File: assets/environment/table_description/package.xml [hunk 3]
```diff
@@ -4,8 +4,8 @@
   <name>table_description</name>
   <version>2.0.0</version>
   <description>table_description description</description>
-  <maintainer email="example@gmail.com">name</maintainer>
-  <license>TODO: License declaration</license>
+  <maintainer email="glenn_tan@artc.a-star.edu.sg">Tan Jing Peng GLenn</maintainer>
+  <license>Apache License 2.0</license>
```
Comment (Briancbn) [inline assets/environment/table_description/package.xml:8]: @cardboardcode  It's already changed here. The link that you reference is to an old commit.

Comment (cardboardcode) [reply assets/environment/table_description/package.xml:8]: Yup. Apologies. Was looking at the outdated file.

**Pull request merged**